### PR TITLE
feature: Support LibreFOMO Integration (Point Localization)

### PR DESCRIPTION
**Pull request by bencejdanko**
# Support LibreFOMO Integration (Point Localization)

## Background issue: 

Located at: 
https://github.com/LibreYOLO/libreyolo/issues/300

## Description

This PR integrates LibreFOMO (Fast Object Localization on Mobile Devices) into LibreYOLO, introducing point localization support. The integration covers the native model definitions, loss functions, custom post-processing, and a test suite. 

Point-based detection is relevant for cases such as robotics, GUI interaction, and document understanding datasets where predicting a central point or click-target is often more practical and contextually accurate than predicting a full bounding box.

PointValidator is added as purely a metric calculator that accepts lists of 2D prediction points, performs bipartite matching (Hungarian assignment) against ground truth, and outputs metrics (Precision, Recall, F1, mean distance). Any future point-localization model must implement three adapter methods:

_decode_point_predictions(...): Translates raw model outputs into a unified list of predicted points [x, y, class, confidence].
_point_metric_shape(...): Tells the validator the spatial dimensions of the target matching space.
_point_targets_from_boxes(...): Defines how dataset labels are projected into the validation matching space.

`point` is added a a task type in `libreyolo/tasks.py`.

## Licensing and Weights

At the moment, the PyTorch tensors are automatically pulled from:

https://huggingface.co/bdanko/LibreFOMO

These were wrapped with this serialization:

```
from libreyolo.utils.serialization import wrap_libreyolo_checkpoint

ckpt = wrap_libreyolo_checkpoint(
    state_dict,
    model_family="librefomo",
    size=size,
    task="point",
    nc=model.nc,
    names={0: "person"},
    imgsz=cfg["imgsz"],
    format_version=2,
    arch="LibreFOMO",
    backbone="mobilenet_v2_keras_tf_same_staticpad_direct_block6_expand",
    alpha=cfg["alpha"],
    head_channels=model.head_channels,
    supported_tasks=["point", "detect"],
    default_task="point",
    state_dict_key="model",
)
```

The full conversion notebook from the original Keras models can be found at:

https://github.com/bencejdanko/librefomo-preparation-and-training/blob/main/convert_keras_mobilenetv2_to_pytorch.ipynb

These are under the cc-by-nc-4.0 (non-commercial) license, due to the original backbone weights being derived from ImageNet training, which require non-commercial usage. In the future, we can add our own pre-trained MobileNetV2 weights from open datasets such as Open Images and COCO. 

Before a final commit, we could possibly relocate from bdanko/LibreFOMO to the official repositories. We can also wait to place anything FOMO related into the official HF repo until we train fully open weights.

## Shared code change

Included is one important location in which I had to touch the Base model behavior at `libreyolo/models/base`:

_rebuild_for_new_classes creates a fresh nn.Module (PyTorch default: training=True) and calls .to(device), but never .eval(). When a checkpoint's nc differs from the factory default (80), this triggers a rebuild during _load_weights. After construction via LibreYOLO(...), model.model.training is True, which is inconsistent with the non-rebuild load path (which correctly calls .eval() at BaseModel.__init__:204). 

I had to add self.model.eval() at the end of _rebuild_for_new_classes. All trainer code explicitly calls model.train() at epoch start, making this idempotent for the training path. 

## Tests

Three test files were added (`@pytest.mark.fomo`):

`tests/unit/test_librefomo.py`:
- Point normalization and grid encoding
- Forward pass output shapes for all three sizes (s/m/l)
- Decode + postprocess: peak extraction, NMS radius, confidence filtering
- `PointsResult` summary and `len()`
- Inference runner wrapping empty (zero-detection) outputs
- `PointValidator` metric computation against known centers

`tests/e2e/test_fomo_inference.py`:
- Checkpoint schema validation: confirms `task='point'`, correct `size`, `imgsz`, `nc` for each of the three pretrained weights (`bdanko/LibreFOMO`)

`tests/e2e/test_fomo_training.py`: 
- Clones `bdanko/sjsu-headcount-scene-1` (YOLO-format, ~8 MB), runs a 2-epoch train of LibreFOMO-s, then verifies loss decreased, `last.pt` exists with a valid checkpoint schema, `metrics/F1` is present in val metrics, and the saved checkpoint reloads via the LibreYOLO factory and returns a valid `result.points` on inference.

## Demonstration Notebook

I have a sample of running the current implementation at this location on a dataset I am using for my purposes:

https://github.com/bencejdanko/librefomo-preparation-and-training/blob/main/train_librefomo_sjsu.ipynb

This conducts a full training run and inference using the usual factory APIs for training and inference, using this `librefomo` branch.

## Limitations

- Multiclass point localization is architecturally supported, but I haven't heavily explored it heavily with a dedicated demo test. My test cases revolve around 1 class, person detection.
- PointValidator currently does point matching occurs in the model output grid space, which may be too FOMO specific, if we want to add point models in the future.
- The point task currently uses a standard detection dataset loader and parses YOLO-format bounding boxes and computes their centroids in _target_points_from_boxes. It does not natively support datasets annotated solely with single points or dot-coordinate labels. This may require a future PointDataset class.

### CODE REVIEW

File: libreyolo/models/librefomo/trainer.py [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+    # -------------------------------------------------------------------------
+    # Setup — build loss and plateau scheduler state
+    # -------------------------------------------------------------------------
+
+    def on_setup(self) -> None:
+        nc = getattr(self.model, "nc", self.config.num_classes)
+        fg_weight = getattr(self.config, "fg_weight", 100.0)
+        self._loss_fn = LibreFOMOLoss(
+            num_classes=nc,
+            fg_weight=fg_weight,
```
[the remark this answers is not in the document]
Comment (bencejdanko) [reply libreyolo/models/librefomo/trainer.py:215]: Will open an issue as this may impact other models

**Comment by bencejdanko** [comment]
Changes are too comprehensive and diverge from the repos main focus on YOLO/RT-DETR, so I don't recommend committing. I'll continue FOMO related work at https://github.com/fomo-edge-ai/fomo if the project wants to support points anytime in the future
